### ISSUE TITLE
Yet another rethink of how the UI could work

### ISSUE THREAD

**Issue by aubergine10** [krzychu124/Cities-Skylines-Traffic-Manager-President-Edition#408]
I'm feeling much more confident about what's possible with UI and interaction model, and while the internet was fubar last night I started pondering if there might be a better overall approach to how the user interaction works.

My idea:

* Activate TMPE
* All segments / nodes become clickable
* Click a segment to access segment tools:
    * Parking restrictions
    * Vehicle restrictions
    * Speed limits
* Click a node to access node tools:
    * Junction restrictions
    * Lane connectors
    * Lane arrows
    * Timed traffic lights

Essentially, you click on the thing you want to customise, then you see the relevant tools. Whatever tool you used last for that type of thing is selected by default, unless that tool is not applicable (eg. train track segments don't have parking restrictions tool) in which case the most commonly used alternative is selected instead.

For example, I click a node and start doing lane arrows. I can then go around the map clicking any node to do it's lane arrows. But the TMPE menu shows the other applicable tools for nodes so I can easily switch to them.

Switching to a different tool retains the currently selected node or segment, for example I could:

* Click segment, edit speed limits
* Switch to parking restrictions, edit those for that same segment
* Switch to vehicle restrictions, edit those for that same segment

The interaction model becomes:

* I, as mayor, want to do something with this node or segment
* So I click it, choose what I want to do, then do it

I'm also factoring in that future versions of the mod are likely to have even more tools. One example is "park and ride" in which user could connect parking buildings to transit hubs, for example to designate a multistory car park as the preferred parking for a train station (search the Issues tab for parking and you'll find a bunch of issues discussing that stuff). As more tools are added, the toolbar is going to get more complex to use, but if users can just activate TM:PE and click something, the toolbar can become context-sensitive showing only the applicable tools for what has been clicked.

**Comment by kvakvs** [comment]
Can create a round menu UI helper for that kind of stuff, and it can remember last state and offer it by default.
![image](https://user-images.githubusercontent.com/288724/60600106-d6cb5c00-9daf-11e9-938f-bd0d8b1ce18f.png)

**Comment by aubergine10** [comment]
Yes. If something is already selected, use TM:PE shortcut (`:`) to pop up the context-sensitive tool wheel. Could possibly also trigger it with middle mouse button, etc.
